api/types/registry/registry.go

```diff
@@ -102,3 +105,15 @@ type SearchResults struct {
 	// Results is a slice containing the actual results for the search
 	Results []SearchResult `json:"results"`
 }
+
+// DistributionInspect describes the result obtained from contacting the
+// registry to retrieve image metadata
+type DistributionInspect struct {
```

Comment:
> I replied to this in the above thread, that the ID only makes sense for schema2 manifests.

This is a somewhat narrow assumption. Every actual runnable image can result in an image id, which is the digest of the image config. I wouldn't recommend exposing ID at all, and just perform a digest resolution.

This functionality is represented with [`Resolver.Resolve`](https://github.com/containerd/containerd/blob/master/remotes/resolver.go#L11) in containerd. It encompasses the action taking a name and resolving to an object with a canonical name. This is an identical action here.